Add eventexecutor workers metric for netty

ttaehee opened the pull request:
### ✏️ Problem
It is difficult to monitor the state of Netty event loop workers in real-time during production, and having this visibility would be helpful for debugging and root cause analysis of network issues.
For example, in our Netty-based WebSocket + STOMP + RabbitMQ environment, we encountered ConnectTimeoutException errors under load. 
Being able to immediately check the status of workers would help us identify and resolve these problems more easily.

### 💡 Solution
Add the eventexecutor.workers metric to expose the number of Netty event loop workers in real-time.

### 📈 Rationale
This metric allows operators to quickly monitor the count and status of workers, enabling faster diagnosis and response to issues. 
Ultimately, it improves system stability and operational efficiency.

### ✅ Local Verification 
I've confirmed locally that the new metric is registered correctly after applying the changes. 
As you can see below, netty.eventexecutor.workers is successfully created with the expected value.

<img width="416" height="199" alt="image" src="https://github.com/user-attachments/assets/b70fb498-eab8-4131-87de-fd270a5e2574" />


related 6375 (실제 pr 시엔 이슈 연결하기)

ttaehee committed:
feat: add workers metric via instanceof check

- it determines the worker count by checking if the provided `Iterable` is an `instanceof MultithreadEventLoopGroup`, allowing it to safely call the `executorCount()` method for an accurate value
- resolve test deprecations

Signed-off-by: ttaehee <ttaehee1105@gmail.com>

injae-kim commented:
Nice work @ttaehee 👍

injae-kim commented on micrometer-core/src/main/java/io/micrometer/core/instrument/binder/netty4/NettyEventExecutorMetrics.java:
```diff
@@ -77,18 +78,26 @@ public NettyEventExecutorMetrics(Iterable<EventExecutor> eventExecutors, Iterabl
         this.eventExecutors.forEach(eventExecutor -> {
             if (eventExecutor instanceof SingleThreadEventExecutor) {
-                SingleThreadEventExecutor singleThreadEventExecutor = (SingleThreadEventExecutor) eventExecutor;
+                SingleThreadEventExecutor singleThreadEventExecutor = (SingleThreadEventExecutor)eventExecutor;
```
이 밑으로 린트때문에 변경된거 같은데, 변경 안할 수 있을까요? 최대한 최소변경으로 PR 올려보고 싶어서요!ㅎ

injae-kim commented on micrometer-core/src/main/java/io/micrometer/core/instrument/binder/netty4/NettyEventExecutorMetrics.java:
```diff
@@ -77,18 +78,26 @@ public NettyEventExecutorMetrics(Iterable<EventExecutor> eventExecutors, Iterabl
     @Override
     public void bindTo(MeterRegistry registry) {
+        if (this.eventExecutors instanceof MultithreadEventLoopGroup) {
+            MultithreadEventLoopGroup eventLoopGroup = (MultithreadEventLoopGroup)this.eventExecutors;
```
```suggestion
            MultithreadEventLoopGroup eventLoopGroup = (MultithreadEventLoopGroup) this.eventExecutors;
```
lint? 로컬에서 린트 체크 한번 돌려주시면 좋을거같습니다!

ttaehee committed:
feat: make workers metric robust for all EventLoopGroup types

- Based on feedback, the 'eventexecutor.workers' metric now supports all EventLoopGroup types, not just MultithreadEventLoopGroup.
- It retains the fast-path for the common case and adds a generic fallback that iterates over the group for an accurate count.
- Tests are expanded to verify all code paths, including subclasses and the new fallback.

Signed-off-by: ttaehee <ttaehee1105@gmail.com>

ttaehee committed:
test: clarify fallback path coverage in NettyEventExecutorMetrics

- Replace shouldHaveWorkersMetricForNonMultithreadEventLoopGroup with a simpler, mock-based test (shouldCountWorkersForGenericIterableViaFallbackPath) to make the fallback path intent explicit

Signed-off-by: ttaehee <ttaehee1105@gmail.com>

ttaehee committed:
fix: prevent strong reference in worker count gauge

- Use WeakReference in worker count gauge to allow executor garbage collection.

Signed-off-by: ttaehee <ttaehee1105@gmail.com>

shakuzen committed:
Polish gauge registration

shakuzen committed:
Polish description
